src/main/java/gregtech/common/covers/filter/BaseFilterContainer.java

```diff
@@ -0,0 +1,170 @@
+package gregtech.common.covers.filter;
+
+import gregtech.api.util.IDirtyNotifiable;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.items.ItemStackHandler;
+
+import com.cleanroommc.modularui.api.widget.IWidget;
+import com.cleanroommc.modularui.screen.ModularPanel;
+import com.cleanroommc.modularui.value.sync.GuiSyncManager;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class BaseFilterContainer<R, T extends Filter<R>> implements INBTSerializable<NBTTagCompound> {
```

Comment:
I'm not sure I like making this an abstract class. I think it would be probably be better to make this a concrete class and try to avoid inheritance if possible.

Reply:
my goal with BaseFilterContainer is to deduplicate code between item and fluid filter containers, since they mostly do the same thing. though, i do see a possibility of taking the item/fluid stack filter-related methods and combining them into a method that accepts an `Object`